### PR TITLE
fix: warnings by shellcheck && cd problem

### PR DESCRIPTION
**Pull request by emig**
- fix following shellcheck warnings:
```
- tests/integration/run-locally.sh|6 col 9 warning| Remove space after = if trying to assign a value (for empty string, use var='' ... ).  [SC1007]
- tests/integration/run-locally.sh|9 col 39 warning| Double quote to prevent globbing and word splitting. [SC2086]
- tests/integration/run-locally.sh|12 col 8 warning| Declare and assign separately to avoid masking return values. [SC2155]
- tests/integration/run-locally.sh|15 col 8 warning| Declare and assign separately to avoid masking return values. [SC2155]
- tests/integration/run-locally.sh|17 col 8 warning| Double quote to prevent globbing and word splitting. [SC2086]
- tests/integration/run-locally.sh|18 col 7 warning| Double quote to prevent globbing and word splitting. [SC2086]
- tests/integration/run-locally.sh|19 col 19 warning| Double quote to prevent globbing and word splitting. [SC2086]
```

- fix `cd` flags problem problem mentioned in #103

Tested in mac, using `bash` and `sh` and `bash --posix`

### CODE REVIEW

File: tests/integration/run-locally.sh [hunk 1]
```diff
@@ -3,19 +3,22 @@
 set -e
 
 expandpath() {
-  CDPATH= cd -- "$1" && pwd -P
+  CDPATH=$(cd -- "$1" && pwd -P)
+  echo "$CDPATH"
 }
```
Comment (badboy) [inline tests/integration/run-locally.sh]: No, this is unsetting `CDPATH`, which might be set by the user. We want to avoid using this in any case.

File: tests/integration/run-locally.sh [hunk 2]
```diff
@@ -3,19 +3,22 @@
...
 
-SCRIPTPATH=$(expandpath "$(dirname -- $0)")
+SCRIPTPATH=$(expandpath "$(dirname -- "$0")")
 cd "$SCRIPTPATH"
 
-export WORKSPACE="$(expandpath ../../..)/workspace"
+export WORKSPACE
+WORKSPACE="$(expandpath ../../..)/workspace"
```
Comment (badboy) [inline tests/integration/run-locally.sh]: Why the split?

Comment (emig) [inline tests/integration/run-locally.sh]: The split resolves warning https://github.com/koalaman/shellcheck/wiki/SC2155

File: tests/integration/run-locally.sh (hunk 1 above)
Comment (emig) [inline tests/integration/run-locally.sh]: Yep, I fix a warning by making the parenthesis also include the second clause, which was wrong. 
If I keep:
```CDPATH=cd -- "$1" && pwd -P```  
I get the warning from shellcheck:
 `To assign the output of a command, use var=$(cmd) . [SC2037]`
but I follow the pattern:
`CDPATH=$(cd -- "$1") && pwd -P`
I don't get cd to the script's directory (probably because it runs in a subshell).

Edge case that can't cover shellcheck?

File: tests/integration/run-locally.sh [hunk 3]
```diff
@@ -3,19 +3,21 @@
 set -e
 
 expandpath() {
-  CDPATH= cd -- "$1" && pwd -P
+  CDPATH=cd -- "$1" && pwd -P
```
Comment (badboy) [inline tests/integration/run-locally.sh]: The space there was intentional. We do not want to assign to `CDPATH`. Another way would be `CDPATH="" cd -- ...`

File: tests/integration/run-locally.sh [hunk 4]
```diff
@@ -3,19 +3,21 @@
...
-export PATH=$(pwd)/bats/bin:$(expandpath ../../target/debug):$PATH
+export PATH
+PATH=$(pwd)/bats/bin:$(expandpath ../../target/debug):$PATH
 
-rm -rf $WORKSPACE
-mkdir $WORKSPACE
-cp -ar fixtures/* $WORKSPACE
+rm -rf "$WORKSPACE"
+mkdir "$WORKSPACE"
+cp -aR fixtures/* "$WORKSPACE"
```
Comment (badboy) [inline tests/integration/run-locally.sh]: Is `-R` different on non-Linux? On my machine the man page says `-r` is the same as `-R`

Comment (emig) [inline tests/integration/run-locally.sh]: in linux:  -a, --archive
              same as -dR --preserve=all

in mac:  -a    Same as -pPR options. Preserves structure and attributes of files but not directory structure.
having -ar in mac causes the incompatiblity -r -R metioned problem.

since  `-R, -r, --recursive copy directories recursively`, 

can we just keep `cp -a`

Comment (emig) [inline tests/integration/run-locally.sh]: my bash version in mac:
`GNU bash, version 3.2.57(1)-release (x86_64-apple-darwin15)`

File: tests/integration/run-locally.sh (hunk 3 above)
Comment (emig) [inline tests/integration/run-locally.sh]: got it, yeah, will make that assignement more explicit.

File: tests/integration/run-locally.sh (hunk 2 above)
Comment (badboy) [inline tests/integration/run-locally.sh]: Interesting. Bash swallows the return value, zsh does not.

File: tests/integration/run-locally.sh (hunk 4 above)
Comment (badboy) [inline tests/integration/run-locally.sh]: `cp -a` should be fine indeed.

**Comment by badboy** [comment]
With the `cp -a` change I think this is now super portable. Will merge #105 asap, but need to fix the test failures first (unrelated to the changes here)
